### ISSUE TITLE
Move elixir version requirement to new metadata field :build_requirements

### ISSUE THREAD

**Issue by ericmj** [hexpm/specifications#20]

**Comment by wojtekmach** [comment]
@ericmj I think that was a decent idea especially if we add a key like minimum_otp_release (that was mentioned elsewhere), so grouping them inside build_requirements map would be neater, but we can wait until we actually add more fields other than elixir requirement.

**Comment by ericmj** [comment]
I agree but at this point it's too late to make changes to the metadata, we need to backwards compatible so a new field would have to be introduced.
